### ISSUE TITLE
Separate the shell normal direction update from ShellStressRelaxationFirstHalf

### ISSUE THREAD

**Issue by WeiyiKong321** [Xiangyu-Hu/SPHinXsys#461]
The normal direction of shell is updated in `ShellStressRelaxationFirstHalf::initialization`, but never used in the relaxation class. It's better to write a separate class for the update of normal direction.

`n_[index_i] = transformation_matrix_[index_i].transpose() * getNormalFromDeformationGradientTensor(F_[index_i]);`

@DongWuTUM Could you please have a look at this issue?

**Comment by DongWuTUM** [comment]
Yes, you can directly move this line to the new class and use the parameters of transformation_matrix_ and F_ in the new class. Could you give it a try first? If you feel confused about how to do it, I'll do it some time.

**Comment by WeiyiKong321** [comment]
> Yes, you can directly move this line to the new class and use the parameters of transformation_matrix_ and F_ in the new class. Could you give it a try first? If you feel confused about how to do it, I'll do it some time.

Ok, I'll do it. Should I create a new branch and PR for this or just edit it in the branch for one-sided fsi?

**Comment by DongWuTUM** [comment]
New branch and PR are better.

**Comment by WeiyiKong321** [comment]
> New branch and PR are better.

Sorry for replying late. I just created the PR https://github.com/Xiangyu-Hu/SPHinXsys/pull/466

**Comment by DongWuTUM** [comment]
Don't need to be sorry :) Thank you.
